app/graphql/mutations/create_song.rb

```diff
@@ -6,31 +6,30 @@ class CreateSong < Mutations::BaseMutation
     field :errors, [String], null: true
 
     def resolve(youtube_id:)
-      persisted_song = Song.find_by(youtube_id: youtube_id)
-      if persisted_song.present?
-        return {
-          song: persisted_song,
-          errors: []
-        }
-      end
+      song = Song.find_or_initialize_by(youtube_id: youtube_id)
+
+      return {
+        song: nil,
+        errors: song.errors.full_messages,
+      } unless song.valid?
 
-      song = Song.new(youtube_id: youtube_id)
-      if song.save
+      unless song.persisted?
         set_attrs_from_youtube!(song)
-        {
-          song: song,
-          errors: [],
-        }
-      else
-        {
-          song: nil,
-          errors: song.errors.full_messages
-        }
+        associate_song_to_user!(song)
       end
+
+      {
+        song: song,
+        errors: [],
+      }
     end
 
     private
 
+    def associate_song_to_user!(song)
+      SongUser.find_or_create_by(song: song, user: context[:current_user])
```

Comment:
The `.find_or_create_by` call here won't actually raise if it fails, it will just [delegate to `create`](https://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by), which [in turn](https://apidock.com/rails/ActiveRecord/Relation/create) just returns the initialized object. I think if this method is supposed to blow up in that case, you gotta bang this call.